happypose/pose_estimators/megapose/inference/icp_refiner.py

```diff
@@ -300,4 +325,4 @@ def refine_poses(
                 predictions_refined.poses[n] = TCO_refined
 
         extra_data = {}
-        return (predictions_refined, extra_data)
+        return (predictions_refined.to(predictions.device), extra_data)
```

Comment:
Can we make `retval` available in the extra_data dictionary?
Something as simple as
```python
extra_data = {
  "return_value": [ retval_1_st_pred, ... retval_n_th_pred ]
}
```

Reply:
To be clearer, what I ask is a 2 or 3 lines change in the code. Feel free not to do it if you have more important things to do.

Reply:
I'm doing it. I was wondering what the retval of `icp_fnc.registerModelToScene` was, according to the sources it seems it's always 0:

https://github.com/opencv/opencv_contrib/blob/ce3c6681c9bf0e5cf46704a6ce0883078bdba074/modules/surface_matching/src/icp.cpp#L466

Reply:
Note that the original idea is to make happypose able to discard detection when ICP fails. Can you also add this ?
This can be done in happypose_ros.

Reply:
Sure, that is my plan.

Reply:
This is done